spark/src/main/scala/geotrellis/spark/costdistance/MrGeoCostDistance.scala

```diff
@@ -0,0 +1,187 @@
+package geotrellis.spark.costdistance
+
+import geotrellis.raster._
+import geotrellis.spark._
+import geotrellis.vector._
+import geotrellis.raster.costdistance.CostDistance
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import org.apache.spark.util.AccumulatorV2
+import org.apache.spark.storage.StorageLevel
+
+import scala.collection.mutable
+
+
+/**
+  * This Spark-enabled implementation of the standard cost-distance
+  * algorithm [1] is "heavily inspired" by the MrGeo implementation
+  * [2] but does not share any code with it.
+  *
+  * 1. Tomlin, Dana.
+  *    "Propagating radial waves of travel cost in a grid."
+  *    International Journal of Geographical Information Science 24.9 (2010): 1391-1413.
+  *
+  * 2. https://github.com/ngageoint/mrgeo/blob/0c6ed4a7e66bb0923ec5c570b102862aee9e885e/mrgeo-mapalgebra/mrgeo-mapalgebra-costdistance/src/main/scala/org/mrgeo/mapalgebra/CostDistanceMapOp.scala
+  */
+object MrGeoCostDistance {
```

Comment:
I think the reference in the doc string is sufficient pointing back to Mr Geo, the type names should probably just be `CostDistance`